### DISCUSSION TITLE
How can I manually supply compiled executables?

### DISCUSSION THREAD

**Discussion by TillWege** [withcatai/node-llama-cpp#72] [General]
The Cmake build system under windows seems rather wonky compared to the the ease of building the llama-cpp repository directly. Where would I have to place the manually compiled executables so that the node runtime would execute them, instead of the pre-built binaries provided by the library?

**Comment by giladgd** [comment]
`llama.cpp` itself uses cmake, so we're also using it here to provide maximum compatibility. From my own experience, cmake works great on Windows.
You can't provide your own executables as `node-llama-cpp` has to build `llama.cpp` by itself to make the bindings work.

`node-llama-cpp` does many things behind the scenes to make the compilation of the `llama.cpp` as smooth as possible when needed.
If you encounter any issues with the compilation, please open an issue with details so we can investigate.
